cabby/model/text/train.py

```diff
@@ -0,0 +1,396 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Simple text classification example.
+
+This script trains a text classifier on a dataset containing paragraphs that
+are linked to either Manhattan or Pittburgh. This is just to get something
+basic that works for the geo-oriented text problems we are addressing with
+the Rendezvous task.
+
+A sentence is encoded with an LSTM and a binary prediction is made from
+the final LSTM state using an MLP.
+
+The data is loaded from raw text files and converted into TF Records.
+
+Example command line call:
+
+$ bazel-bin/cabby/model/text/train \
+  --data_dir ~/data/morp/morp-balanced  \
+  --model_dir ~/tmp/morp-model
+
+This code is copied with very few modifications from the Flax SST2 example:
+
+https://github.com/google/flax/tree/master/examples/sst2
+"""
+# pylint: disable=import-error,too-many-locals,too-many-arguments
+
+import collections
+from typing import Any, Dict, Text, Tuple
+
+from absl import app
+from absl import flags
+from absl import logging
+
+import flax
+import flax.training.checkpoints
+from flax import nn
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets as tfds
+from tensorflow.io import gfile
+
+import input_pipeline
+import model as sst2_model
+
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_float(
+    'learning_rate', default=0.0005,
+    help=('The learning rate for the Adam optimizer.'))
+
+flags.DEFINE_integer(
+    'batch_size', default=64,
+    help=('Batch size for training.'))
+
+flags.DEFINE_integer(
+    'num_epochs', default=20,
+    help=('Number of training epochs.'))
+
+flags.DEFINE_float(
+    'dropout', default=0.5,
```

Comment:
wow this looks like a very high value for dropout! may be better to have a lower default.